backstage/plugins/budget-usage-backend/src/bigquery.ts

```diff
@@ -166,51 +173,93 @@ export async function saveBudgetUsages(budgetUsages: Usage[]) {
 }
 
 /**
- * Fetches new budgets from the billing account and syncs them with BigQuery.
- *
- * @throws Will throw an error if there is an issue fetching or syncing the budgets.
+ * Fetches and syncs new budgets from the budget client and inserts them into a BigQuery table if they don't already exist.
  */
 export async function fetchAndSyncNewBudgets() {
+  // Get the dataset and table references from the config
+  const dataset = bigqueryClient.dataset(config.bigquery.budgets.dataset);
+  const table = dataset.table(config.bigquery.budgets.tables.budget.name);
+
   try {
+    // DEBUGGING SA used to make requests
+    const saToken = await bigqueryClient.authClient.getCredentials();
+    console.log('SA EMAIL:', saToken.client_email);
+
+    // Fetch the list of budgets
     const [budgets] = await budgetClient.listBudgets({
       parent: config.billing.accountId,
     });
 
-    const dataset = bigqueryClient.dataset(config.bigquery.budgets.dataset);
-    const table = dataset.table(config.bigquery.budgets.tables.budget.name);
-
+    // Process each budget
     for (const budget of budgets) {
-      const [queryResult] = await table.query({
-        query: checkBudgetExistsQuery,
-        params: {
-          Id: budget.name,
-        },
-      });
-      if (queryResult?.length === 0) {
-        table.insert(
-          [
-            {
-              name: budget.name,
-              projectId: budget.displayName,
-              amount: budget.amount?.specifiedAmount?.units
-                ? parseFloat(String(budget.amount.specifiedAmount.units))
-                : 0,
-              currencyCode: budget.amount?.specifiedAmount?.currencyCode,
-            },
-          ],
-          err => {
-            if (err) {
-              console.error(
-                'error registering new project' + budget.displayName,
-              );
-            }
-          },
-        );
+      try {
+        // Check if the budget already exists in the table
+        const budgetExists = await checkIfBudgetExists(table, budget.name!);
+        // If the budget does not exist, insert it
+        if (!budgetExists) {
+          console.log(`Inserting new budget: ${budget.displayName}`);
+          await insertBudget(table, budget);
+        }
+      } catch (error) {
+        // Log error for the specific budget being processed
+        console.error(`Error processing budget ${budget.displayName}:`, error);
       }
     }
   } catch (error) {
-    console.error(error);
-    throw new Error(`Error syncing budget`);
+    // Log error for the entire budget fetching and syncing process
+    console.error('Error fetching and syncing budgets:', error);
+    throw error;
+  }
+}
+
+/**
+ * Checks if a budget with the given ID exists in the specified table.
+ * @param {object} table - The BigQuery table object.
+ * @param {string} budgetId - The ID of the budget to check.
```

Comment:
**Non-blocking**: The param name doesn't match. It'll be caught by a linting rule that isn't set up yet. Personally I'd remove anything that's obvious/redundant to avoid stuff like this.